lib/fastly_nsq/listener.rb

```diff
@@ -113,21 +113,26 @@ def initialize(topic:, processor:, preprocessor: FastlyNsq.preprocessor, channel
   def call(nsq_message)
     message = FastlyNsq::Message.new(nsq_message)
 
+    msg_info = {
+      channel:  channel,
+      topic:    topic,
+      attempts: nsq_message.attempts,
+      id:       Digest::MD5.hexdigest(nsq_message.body.to_s),
+      metadata: message.meta,
+    }
+
     logger.info do
-      {
-        channel:  channel,
-        topic:    topic,
-        attempts: nsq_message.attempts,
-        id:       Digest::MD5.hexdigest(nsq_message.body.to_s),
-        metadata: message.meta,
-      }.tap do |l|
+      msg_info.tap do |l|
         l[:data] = message.body if logger.level == Logger::DEBUG
       end
     end
 
-    preprocessor&.call(message)
-    result = processor.call(message)
-    message.finish if result
+    FastlyNsq.tracer.trace_with_newrelic(msg_info) do
+      preprocessor&.call(message)
+      result = processor.call(message)
+      message.finish if result
+    end
```

Comment:
Do we want to trace every `processor.call`? I'm wondering about any potential performance issues.

Reply:
I know it's in a background thread etc etc, just wondering

Reply:
as opposed to random sampling?  I think we do want every processor call traced, since data values/differences can change the characteristics of what happens.

Reply:
I was just thinking about massive backfills and if we are adding 5% overhead to every `processor.call` it might be something to think about or be able to toggle.

Reply:
@set5think per our discussion I think this is good to add around every call. A next step might be a toggle that allows listeners to determine when to trace. Would be cool to have it be`on/off/sample`

Reply:
yea that's a nice idea.